### PR TITLE
feat: added marketplace buttons and markdown renderer

### PR DESCRIPTION
**Pull request by ssikande**
added custom buttons that will be imported from the extension
image

- - -
This change is [Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1009)

### CODE REVIEW

**Review by jolierabideau — CHANGES_REQUESTED** [review]
Thanks for working on this Sameer!

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1009#-:-O20ibrFD6bdblfhr8E1:b-qu2cln)* status: all files reviewed, 21 unresolved discussions (waiting on @ssikande)

___
*[`lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx` line 5 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20dAFO2bvjc7tePkZZ:-O20dAFO2bvjc7tePkZ_:b-vnuueb) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx#L5)):*
> ```XML
> interface MarkdownRendererProps {
>     markdown: string;
>     }
> ```

This shouldn't be indented, do you need to `npm run lint` or `npm run build`?
___
*[`lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx` line 10 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20dKPD2iJrs-takyO9:-O20dKPD2iJrs-takyOA:bnu4c8r) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx#L10)):*
> ```XML
>     return (
>         <div className="pr-prose">
>             <Markdown options={{}}>{markdown}</Markdown>
> ```

Sorry if I asked this before but do you have to provide an empty options object? Or can you just remove that prop?
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 4 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20g9PV0b6QFCGetR4L:-O20g9PW1RAzGTjsIvwd:b-hcsr8) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L4)):*
> ```XML
> import { Download, LoaderCircle } from 'lucide-react';
> 
> interface DownloadButtonProps {
> ```

It would be helpful to add a /** */ comment for each of the props here, so that it shows up on hover when people use this component. You could probably use the ones you added in the function TSDoc. I also think we want this to be `type DownloadButtonProps` instead, is there a specific reason why you used an interface? I can't remember from the first time I reviewed
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 5 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20dTXxAs3eszKtPXEh:-O20dTXxAs3eszKtPXEi:ba4qdw2) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L5)):*
> ```XML
> 
> interface DownloadButtonProps {
>   downloading: boolean;
> ```

Can this be `isDownloading` as requested by the code style guide
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 7 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hKfW8EjpQ8V_DuvA:-O20hKfW8EjpQ8V_DuvB:brjw3gr) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L7)):*
> ```XML
>   downloading: boolean;
>   handleClick: () => void;
>   position: 'left' | 'right';
> ```

Can this prop be called `iconPosition`?
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 12 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20drBX-JJc1YyGWNvh:-O20drBX-JJc1YyGWNvi:b-d4v245) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L12)):*
> ```XML
> 
> /**
>  * This is a download button component that provides the styling and props for a download button.
> ```

Can we make this more descriptive? Like "The `DownloadButton` component is a button designed for initiating downloads. It includes visuals for active downloading and idle states."
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 28 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20fBCI6sv1URbITsmc:-O20fBCJD8j6miNehMrI:bmh3f1o) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L28)):*
> ```XML
>   return (
>     <Button
>       className={`pr-h-8 pr-w-20 pr-rounded-md pr-px-4 pr-text-white pr-transition pr-duration-300 pr-ease-in-out ${
> ```

When we are concatenating `classNames` we want to use the `cn` utility function. I don't think you had access to that in the extension, but there are lots of examples of components using this in PBR. I pasted an example from `BookMenuItem`. When adding conditionals to the `cn` function, we are trying to follow this object pattern:

```typescript
className={cn('pr-font-normal pr-text-slate-700', {
    'pr-bg-amber-50 pr-text-yellow-900 data-[highlighted]:pr-bg-amber-100': isSelected,
})}
```
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 41 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20fqI04ZnOS49spFXl:-O20fqI04ZnOS49spFXm:bzg5ove) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L41)):*
> ```XML
>         <>
>           {position === 'right' && <Download size={25} className="pr-mr-2 pr-h-4 pr-w-4" />}
>           {position === 'right' && buttonText}
> ```

Regardless what the position is we want the buttonText to render here, right? So this could just be `{buttonText}`
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 4 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hSz-86sFX5btrF-j:-O20hSz-86sFX5btrF-k:b-z9m9v1) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L4)):*
> ```XML
> import { LoaderCircle } from 'lucide-react';
> 
> interface RemoveButtonProps {
> ```

Same comment about types vs interfaces and adding /** */ comments to these props
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 5 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hPrl5Qa43noqsGY3:-O20hPrl5Qa43noqsGY4:b-pvp8uj) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L5)):*
> ```XML
> 
> interface RemoveButtonProps {
>   removing: boolean;
> ```

Can this be called `isRemoving` per code style guide
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 10 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20git58qexCNvuC4Pw:-O20git58qexCNvuC4Px:bmjz6vd) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L10)):*
> ```XML
> 
> /**
>  * This is an remove button component and provides styling and props for an remove button.
> ```

Same comment I made on DownloadButton- a more descriptive comment here would be helpful
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 16 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20gyZh4ZHaL0ybbdCj:-O20gyZh4ZHaL0ybbdCk:b7tjfty) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L16)):*
> ```XML
>  * @returns A button that can be used to remove.
>  */
> export default function RemoveButton({ removing, handleClick }: RemoveButtonProps) {
> ```

Should this component also have a `iconPosition`?
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 19 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20gqWJ-7gcKQ_bN1ku:-O20gqWJ-7gcKQ_bN1kv:bf5x41n) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L19)):*
> ```XML
>   return (
>     <Button
>       className={`pr-h-8 pr-rounded-md pr-bg-gray-300 pr-px-4 pr-text-black pr-transition pr-duration-300 pr-ease-in-out hover:pr-text-white ${removing ? 'pr-cursor-not-allowed pr-bg-gray-400' : ''}`}
> ```

Same comment here about using `cn` and the object pattern for conditionals
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 4 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hcSX31zWUbwlClit:-O20hcSX31zWUbwlCliu:bctyt9x) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L4)):*
> ```XML
> import { LoaderCircle } from 'lucide-react';
> 
> interface UpdateButtonProps {
> ```

Same comment about type and comments on props
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 5 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hgS10No8heHUg7AI:-O20hgS26ggzLb7VxhMz:b7zfm5x) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L5)):*
> ```XML
> 
> interface UpdateButtonProps {
>   updating: boolean;
> ```

`isUpdating`
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 10 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hhrq2_7wZ6rRU3LJ:-O20hhrq2_7wZ6rRU3LK:b5wrjvt) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L10)):*
> ```XML
> 
> /**
>  * This is an update button component and provides styling and props for an update button.
> ```

Same comment about more descriptive description :)
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 16 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hr5_8dMA6a9bYwOB:-O20hr5_8dMA6a9bYwOC:b-zgooas) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L16)):*
> ```XML
>  * @returns A button that can be used to update.
>  */
> export default function UpdateButton({ updating, handleClick }: UpdateButtonProps) {
> ```

Should we have an `iconPosition`?
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 19 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20hnO7AJN-3a-ulWbf:-O20hnO7AJN-3a-ulWbg:b-amkzlb) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L19)):*
> ```XML
>   return (
>     <Button
>       className={`pr-h-8 pr-rounded-md pr-bg-blue-600 pr-px-4 pr-text-white pr-transition pr-duration-300 pr-ease-in-out hover:pr-text-white ${updating ? 'pr-cursor-not-allowed pr-bg-blue-700' : ''}`}
> ```

`cn` function :)
___
*[`lib/platform-bible-react/src/preview/components/compositions.component.tsx` line 6 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20i1rv6iY8xsuUOiza:-O20i1rw4PJdZ3l5dPFc:ba1tobk) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/preview/components/compositions.component.tsx#L6)):*
> ```XML
> import DownloadButton from '@/components/extension-marketplace/buttons/download-button.component';
> import RemoveButton from '@/components/extension-marketplace/buttons/remove-button.component';
> import UpdateButton from '@/components/extension-marketplace/buttons/update-button.component';
> ```

Nice job using path aliases!!
___
*[`lib/platform-bible-react/src/preview/components/compositions.component.tsx` line 39 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20iDrAA2nPuxiiAVfp:-O20iDrAA2nPuxiiAVfq:bmm1brk) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/preview/components/compositions.component.tsx#L39)):*
> ```XML
>           </VerticalTabsTrigger>
>           <VerticalTabsTrigger value="Theme Toggle">Theme Toggle</VerticalTabsTrigger>
>           <VerticalTabsTrigger value="Marketplace Buttons">Marketplace Buttons</VerticalTabsTrigger>
> ```

I think it would be a good idea to go ahead and add a tab for each of the buttons, because they are going to be used outside of the marketplace. 
___
*[`lib/platform-bible-react/src/preview/components/compositions.component.tsx` line 56 at r1](https://reviewable.io/reviews/paranext/paranext-core/1009#-O20iNNMBU_BKLgC5fz5:-O20iNNMBU_BKLgC5fz6:b9nns6d) ([raw file](https://github.com/paranext/paranext-core/blob/a6fae3c4d4bfe45d8f19c0cb003342bb5d71e7cc/lib/platform-bible-react/src/preview/components/compositions.component.tsx#L56)):*
> ```XML
>         </VerticalTabsContent>
> 
>         <VerticalTabsContent value="Marketplace Buttons">
> ```

Nice job using them with varying props!! A little nitpicky but can we get the buttons to appear in line with each other? And maybe add some spacing between them?

File: lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import Markdown from 'markdown-to-jsx';
+
+interface MarkdownRendererProps {
+    markdown: string;
+    }
```
Comment (captaincrazybro) [inline lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx:5]: Should this extra tab be here?

File: lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export default function MarkdownRenderer({ markdown }: MarkdownRendererProps) {
```
Comment (captaincrazybro) [inline lib/platform-bible-react/src/components/extension-marketplace/markdown-renderer.component.tsx:7]: TSDOC

**Review by captaincrazybro — CHANGES_REQUESTED** [review]
Looks great! Just two comments.

**Review by jolierabideau — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1009)* status: all files reviewed, 8 unresolved discussions (waiting on @ssikande)

___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx` line 39 at r2](https://reviewable.io/reviews/paranext/paranext-core/1009#-O21AkjiB2sivHv3qvCt:-O21AkjiB2sivHv3qvCu:b7ye3lz) ([raw file](https://github.com/paranext/paranext-core/blob/060c9ec07645c01f6dc9f08f25313c0c96ffa4c5/lib/platform-bible-react/src/components/extension-marketplace/buttons/download-button.component.tsx#L39)):*
> ```XML
>         'pr-h-8 pr-rounded-md pr-text-white pr-transition pr-duration-300 pr-ease-in-out',
>         isDownloading ? 'pr-bg-blue-700 pr-cursor-not-allowed' : 'pr-bg-blue-600',
>         !buttonText ? 'pr-bg-white pr-text-blue-600' : 'pr-w-20',
> ```

See how the other components do conditionals, there is an object pattern that you can use with the `cn` function. 

```
{ 'className': isDownloading, 'className': !isDownloading, .....add more }
```
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx` line 28 at r2](https://reviewable.io/reviews/paranext/paranext-core/1009#-O21BFf2BrWS_0MEK-Fc:-O21BFf2BrWS_0MEK-Fd:b5xbp2w) ([raw file](https://github.com/paranext/paranext-core/blob/060c9ec07645c01f6dc9f08f25313c0c96ffa4c5/lib/platform-bible-react/src/components/extension-marketplace/buttons/remove-button.component.tsx#L28)):*
> ```XML
>       className={cn(
>         'pr-h-8 pr-rounded-md pr-bg-gray-300 pr-text-black pr-transition pr-duration-300 pr-ease-in-out hover:pr-text-white',
>         isRemoving ? 'pr-cursor-not-allowed pr-bg-gray-400' : '',
> ```

Same comment about object pattern
___
*[`lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx` line 28 at r2](https://reviewable.io/reviews/paranext/paranext-core/1009#-O21BQxu1vHiRzIV2IX7:-O21BQxu1vHiRzIV2IX8:b5xbp2w) ([raw file](https://github.com/paranext/paranext-core/blob/060c9ec07645c01f6dc9f08f25313c0c96ffa4c5/lib/platform-bible-react/src/components/extension-marketplace/buttons/update-button.component.tsx#L28)):*
> ```XML
>       className={cn(
>         'pr-h-8 pr-rounded-md pr-bg-blue-600 pr-px-4 pr-text-white pr-transition pr-duration-300 pr-ease-in-out hover:pr-text-white',
>         isUpdating ? 'pr-cursor-not-allowed pr-bg-blue-700' : '',
> ```

Same comment about object pattern

**Review by jolierabideau — DISMISSED** [review]
Nice work!

> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1009#-:-O21EWWp-67mXY5w-etB:bdtx8r6)* status: all files reviewed, 2 unresolved discussions (waiting on @ssikande)

**Review by captaincrazybro — DISMISSED** [review]
LGTM!

**Review by jolierabideau — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1009)* status: all files reviewed, 2 unresolved discussions (waiting on @ssikande)

**Review by captaincrazybro — APPROVED** [review]

**Pull request merged**